SCE-Development/Core-v4
# Add Login with Discord button

# SCE Authentication with Discord

Implements a login button to connect your SCE account to your Discord account.

# How to test
- in `api/config`, locate `config.example.json` and create a copy of that file, renaming it to `config.json`
- in this new file created, obtain `CLIENT_ID` and `CLIENT_SECRET` from sce-dev discord, then paste them in the respective fields (this is in `discordApiKeys`)
- make sure mongodb is running, then in the terminal, open the server and frontend (`npm run server` then `npm start`)
- login with sce account, navigate to the profile page, you'll see the page look like this
![image](https://user-images.githubusercontent.com/66745211/89147267-00067080-d50b-11ea-9bb9-c73b1afd9431.png)
- click on "connect with discord" and follow the prompts. the authentication page will open in a new tab
- login with your discord account, then close the discord tab if it displays "Authorization successful! You may now close this window or tab"
- refresh the profile page. you'll see the page now looking like this 
![image](https://user-images.githubusercontent.com/66745211/89147346-35ab5980-d50b-11ea-9a88-bc28d32e4b26.png)
- note! if you have a discord account linked already, the profile screen will appear as it does above. since a person's username (and discriminator, with discord nitro) can be changed at will, it is recommended that each time that person updates said information, they also update with sce their new username (and discriminator). discordID will not change.

# Final notes

- upon the return of the official sce website, instead of redirecting to discord authorization page, it will instead redirect back to the profile page.
